tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/firebase/FirebaseConfiguration.java

```diff
@@ -1,4 +1,51 @@
 package com.linagora.tmail.james.jmap.firebase;
 
-public record FirebaseConfiguration(String firebasePrivateKeyUrl) {
+import java.util.Optional;
+
+import org.apache.commons.configuration2.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linagora.tmail.james.jmap.model.MissingOrInvalidFirebaseCredentialException;
+
+public record FirebaseConfiguration(String firebasePrivateKeyUrl,
+                                    Optional<String> apiKey,
+                                    Optional<String> appId,
+                                    Optional<String> messagingSenderId,
+                                    Optional<String> projectId,
+                                    Optional<String> databaseUrl,
+                                    Optional<String> storageBucket,
+                                    Optional<String> authDomain,
+                                    Optional<String> vapidPublicKey) {
```

Comment:
In a perfect world we would get strong types in our configuration POJO

CF 

![image](https://github.com/user-attachments/assets/0060a716-bdd1-48f1-b6b6-53968bd07574)

Reply:
I hesitated with but it just complexifies the code for not so much honestly

Reply:
as you wish